### ISSUE TITLE
docs do not include a SnarkyJS codebase overview

### ISSUE THREAD

**Issue by barriebyron** [o1-labs/docs2#372]
By providing a SnarkyJS codebase overview in our docs, we help our zkApp developers, builders, and contributors.

In addition to helpful inline code comments, a codebase overview describes the important parts of the codebase. For this issue, let's create an overview of the SnarkyJS codebase and describe how the pieces fit together. 

This issue is related to a larger, living, and ongoing effort to reduce friction and improve developer experience, thanks @shimkiv for https://github.com/o1-labs/docs2/issues/356 

It's helpful to know how the https://docs.minaprotocol.com/zkapps/snarkyjs-reference docs are generated:
- https://github.com/o1-labs/snarkyjs/blob/main/src/lib/circuit_value.ts generates 
https://docs.minaprotocol.com/zkapps/snarkyjs-reference/classes/CircuitValue 

- https://github.com/o1-labs/snarkyjs/blob/main/src/lib/mina.ts  generates https://docs.minaprotocol.com/zkapps/snarkyjs-reference/modules/Mina#type-declaration-2 

**Examples of codebase overviews**
Mina > Node Developers > [Codebase Overview](https://docs.minaprotocol.com/node-developers/codebase-overview) and [Repository Structure](https://docs.minaprotocol.com/node-developers/repository-structure)

Other examples in the wild can provide inspiration, including:
- [Solidity](https://docs.soliditylang.org/en/latest/index.html) docs, including [Layout of a Solidity Source File](https://docs.soliditylang.org/en/latest/layout-of-source-files.html) 
- Super high level [Polygon zkEVM Source Code](https://wiki.polygon.technology/docs/zkEVM/open-source-repos/)

**Comment by jasongitmail** [comment]
I agree with the value here, it's something we've discussed before too because I think it'd be useful for contributors. But I see this better to live in the SnarkyJS repo because it's specific to _contributing to SnarkyJS_, as opposed to using SnarkyJS for writing a zkApp

**Comment by barriebyron** [comment]
@jasongitmail  thank you for moving the issue. I inadvertently created the issue in the wrong repo... and TIL issues can be moved to another repo. (How do I do this?)
